### DISCUSSION TITLE
Provider Resource to Manage Grant on COMMAND

### DISCUSSION THREAD

**Discussion by mswanson-mastery** [Snowflake-Labs/terraform-provider-snowflake#3010] [Q&A] [answered]
Hey folks, 

We are onboaring a data governance tool and one of the permissions it requires is usage on the `SHOW IMPORTED KEYS` command - I cannot find a single provider def in the docs to manage this - other than an unsafe execute. 

We have no qualms about using that to grant usage on the command, but I'm curious if there are other definitions I could use to accomplish this?

Thanks!

**Comment by sfc-gh-jcieslak** [comment]
Hey @mswanson-mastery 👋 
To be honest, that's the first time I've seen this command/privilege. Just so you know, we are basing our knowledge on the public documentation, so If anything is missing, we won't have it in the provider (I already reported it, and it should be added to the [official documentation](https://docs.snowflake.com/)). Could you provide an SQL example of how you would like to grant usage on that command? As a workaround for now, you can use [unsafe_execute](https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/resources/unsafe_execute) that uses raw SQL.

**Reply by sfc-gh-jcieslak** [reply]
Hey @mswanson-mastery 
SHOW IMPORTED KEYS is planned to be added to official documentation, but I'm still waiting for the answer from other teams if it's possible to grant usage on this function.

**Reply by sfc-gh-jcieslak** [reply]
Sorry, but I didn't get any response regarding granting usage for SHOW IMPORTED KEYS. I'm guessing you can get a faster response by reporting this to Snowflake directly.

**Reply by mswanson-mastery** [reply, marked as answer]
Hello Jan! 

Thank you for the response, I actually forgot I opened this issue and reached out to the provider of the tool itself, they told us that we could ignore the grant of access to the command, so I was unblocked after that. 
Apologies for the open loop. 

-Matt
